scipy/special/basic.py

```diff
@@ -2462,3 +2462,61 @@ def zeta(x, q=None, out=None):
         q = 1
     return _zeta(x, q, out)
 
+
+def owens_t(h, a):
+    r"""
+    Owen's T function.
+
+    Parameters
+    ----------
+    h: int or float
+        Input value.
+    a: int or float
+        Input value.
+        If ``a == 0``, the return value is 0.
+
+    Returns
+    -------
+    t: float
+        Probability of the event (X > h and 0 < Y < a * X),
+        where X and Y are independent standard normal random variables.
+
+    References
+    ----------
+    .. [1] Mike Patefield and David Tandy. "Fast and accurate calculation
+           of Owen's T-function", 2000.
+           https://www.jstatsoft.org/article/view/v005i05/t.pdf
+
+    """
+    if not (np.isscalar(h) and np.isscalar(a)):
+        raise ValueError("arguments must be scalars.")
+    if h < 0:
+        return owens_t(-h, a)
+    if a < 0:
+        return -owens_t(h, -a)
+    elif a > 1:
+        ncdf_h = ndtr(h)
+        ncdf_ah = ndtr(a * h)
+        return 0.5 * (ncdf_h + ncdf_ah) - ncdf_h * ncdf_ah -\
+            owens_t(a * h, 1 / a)
+
+    if a == 0:
+        return 0
+    if h == 0:
+        return 1 * np.arctan(a) / (2 * np.pi)
+    if a >= 0.99:
+        ncdf = ndtr(h)
+        return 0.5 * ncdf * (1 - ncdf)
+
+    result = 0
+    z_prev = 0
+    z_curr = (ndtr(a * h) - 0.5) / h
+    ah_sq = -0.5 * h * h * a * a
+    pi_sq = np.power(2 * np.pi, -0.5)
+
+    for i in xrange(int(a * a * h * h / 2) + 1):
+        result += ((-1) ** (i - 1)) * z_curr
+        z_prev = z_curr
+        z_curr = ((2 * i - 1) * z_prev - pi_sq *
+            np.power(a, 2 * i - 1) * np.exp(ah_sq)) / (h * h)
```

Comment:
and style nitpick: lines like these should line up with the opening bracket above, not use 4-space indents